activation/e2e/activation_test.go

```diff
@@ -33,50 +34,41 @@ import (
 	"github.com/spacemeshos/go-spacemesh/timesync"
 )
 
-func Test_BuilderWithMultipleClients(t *testing.T) {
-	ctrl := gomock.NewController(t)
-
-	const numEpochs = 3
-	const numSigners = 3
-	const totalAtxs = numEpochs * numSigners
-
-	signers := make(map[types.NodeID]*signing.EdSigner, numSigners)
-	for range numSigners {
-		sig, err := signing.NewEdSigner()
-		require.NoError(t, err)
-
-		signers[sig.NodeID()] = sig
-	}
-
-	logger := zaptest.NewLogger(t)
-	goldenATX := types.ATXID{2, 3, 4}
-	cfg := activation.DefaultPostConfig()
-	db := sql.InMemory()
-	localDB := localsql.InMemory()
-
+func syncedSyncer(ctrl *gomock.Controller) *activation.Mocksyncer {
 	syncer := activation.NewMocksyncer(ctrl)
 	syncer.EXPECT().RegisterForATXSynced().DoAndReturn(func() <-chan struct{} {
 		synced := make(chan struct{})
 		close(synced)
 		return synced
 	}).AnyTimes()
+	return syncer
+}
 
+func initializeIDs(
```

Comment:
I'm not a big fan of passing multiple signers to one init function for the sake of parallelisation. Also for the caller it is not clear that the `opts` are changed for each identity without looking into the implementation. Instead I'd prefer the previous approach where `initPost` is called once per identity.

If you want to reduce the amount of code needed per identity I would instead move some of the setup code for `initPost` into the function instead of having one function to initialise multiple identities.

For instance the `initPost` function could be extended like this:

```go
func initPost(
	tb testing.TB,
	ctrl *gomock.Controller,
	logger *zap.Logger,
	db sql.Executor,
	cfg activation.PostConfig,
	opts activation.PostSetupOpts,
	svc *grpcserver.PostService,
	grpcCfg grpcserver.Config,
	goldenATX types.ATXID,
	sig *signing.EdSigner,
) {
	tb.Helper()

	validator := activation.NewMocknipostValidator(ctrl)
	syncer := syncedSyncer(ctrl)
	mgr, err := activation.NewPostSetupManager(cfg, logger, db, atxsdata.New(), goldenATX, syncer, validator)
	require.NoError(tb, err)

	// Create data.
	require.NoError(tb, mgr.PrepareInitializer(context.Background(), opts, sig.NodeID()))
	require.NoError(tb, mgr.StartSession(context.Background(), sig.NodeID()))
	require.Equal(tb, activation.PostSetupStateComplete, mgr.Status().State)

	tb.Cleanup(launchPostSupervisor(tb, logger, mgr, sig, grpcCfg, opts))

	require.Eventually(tb, func() bool {
		_, err := svc.Client(sig.NodeID())
		return err == nil
	}, 10*time.Second, 100*time.Millisecond, "timed out waiting for connection")
}
```

And then in the test with multiple identities the loop to set them all up would be:

```go
	var eg errgroup.Group
	i := uint32(1)
	opts := testPostSetupOpts(t)
	for _, sig := range signers {
		opts := opts
		opts.DataDir = t.TempDir()
		opts.NumUnits = min(cfg.MinNumUnits+i*2, cfg.MaxNumUnits)
		i += 1
		eg.Go(func() error {
			initPost(t, ctrl, logger, db, cfg, opts, svc, grpcCfg, goldenATX, sig)
			return nil
		})
	}
	require.NoError(t, eg.Wait())
```

Putting the configuration in the callers hands (clearer what is happening) and hide the "boring" setup stuff in the `initPost` function.